src/murmur/Server.cpp

```diff
@@ -1464,17 +1465,19 @@ void Server::message(unsigned int uiType, const QByteArray &qbaMsg, ServerUser *
 
 		MessageHandler::UDPMessageType msgType = static_cast<MessageHandler::UDPMessageType>((buffer[0] >> 5) & 0x7);
 
-		switch (msgType) {
-			case MessageHandler::UDPVoiceCELTAlpha:
-			case MessageHandler::UDPVoiceCELTBeta:
-			case MessageHandler::UDPVoiceSpeex:
-				if (bOpus)
-					break;
-			case MessageHandler::UDPVoiceOpus:
-				processMsg(u, buffer, l);
-				break;
-			default:
-				break;
+		if (msgType == MessageHandler::UDPVoiceSpeex ||
+			msgType == MessageHandler::UDPVoiceCELTAlpha ||
+			msgType == MessageHandler::UDPVoiceCELTBeta ||
+			msgType == MessageHandler::UDPVoiceOpus) {
+
+			// Determine if the packet should be processed
+			// by Murmur. Clients can still send non-Opus
+			// voice packets when in Opus mode. However,
+			// Murmur should discard them in that case.
+			if (bOpus && msgType != MessageHandler::UDPVoiceOpus) {
```

Comment:
This logic doesn't make sense? "If we're in Opus mode, only process packets that are not Opus".

Reply:
See my previous comment.